src/patterns/ec2-app.ts

```diff
@@ -5,21 +5,48 @@ import type { GuCertificateProps } from "../constructs/acm";
 import { GuCertificate } from "../constructs/acm";
 import type { GuUserDataProps } from "../constructs/autoscaling";
 import { GuAutoScalingGroup, GuUserData } from "../constructs/autoscaling";
-import { Gu5xxPercentageAlarm } from "../constructs/cloudwatch";
 import type { Gu5xxPercentageMonitoringProps, NoMonitoring } from "../constructs/cloudwatch";
+import { Gu5xxPercentageAlarm } from "../constructs/cloudwatch";
 import type { GuStack } from "../constructs/core";
 import { AppIdentity } from "../constructs/core/identity";
-import { GuVpc, SubnetType } from "../constructs/ec2";
+import { GuSecurityGroup, GuVpc, SubnetType } from "../constructs/ec2";
 import { GuGetPrivateConfigPolicy, GuInstanceRole } from "../constructs/iam";
 import {
   GuApplicationListener,
   GuApplicationLoadBalancer,
   GuApplicationTargetGroup,
 } from "../constructs/loadbalancing";
+import { transformToSecurityGroupAccessRule } from "../utils/security-groups";
+
+const PUBLIC = "PUBLIC" as const;
+const INTERNAL = "INTERNAL" as const;
+const RESTRICTED = "RESTRICTED" as const;
+
+interface Access {
+  type: string;
+}
+
+// Everyone on the internet can access
+interface PublicAccess extends Access {
+  type: typeof PUBLIC;
+}
+
+// Access restricted to internal AWS networks
```

Comment:
As a developer setting up a new app, I'm unsure when I would want to use `INTERNAL` and when I'd want to use `RESTRICTED`, perhaps we can try to clarify that here?